Parser/tokenizer.c

```diff
@@ -874,8 +874,12 @@ tok_nextc(struct tok_state *tok)
                 strcpy(newtok, buf);
                 Py_DECREF(u);
             }
-            if (tok->nextprompt != NULL)
-                tok->prompt = tok->nextprompt;
+            if (tok->nextprompt != NULL){
+                if (*newtok == '#' || *newtok == ' ' || *newtok == '\t')
```

Comment:
This can cause a segfault if `newtok` is `NULL`, and that can be the case -- see the next check. To repro, hit ^C at the prompt.